transports/transports.go

```diff
@@ -77,7 +77,9 @@ func ListNames() []string {
 	defer kt.mu.Unlock()
 	var names []string
 	for _, transport := range kt.transports {
-		names = append(names, transport.Name())
+		if !transport.Deprecated() {
```

Comment:
Can be actually depreciating transports in this package directly by filtering them out here? Cluttering the interface with a deprecated method doesn't sound right to me. @mtrmac what do you think? Transport is an interface and depreciating it in the library means just documenting that it's deprecated. Users of the interface (this package I'm commenting on) should filter what they think it's deprecated.